lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -746,30 +710,30 @@ utxoFromTxOuts era = withConstraints era $
 utxoFromTxOutsInRecentEra
     :: forall era. RecentEra era
     -> [(TxIn, TxOutInRecentEra)]
-    -> Either ErrInvalidTxOutInEra (Shelley.UTxO (ShelleyLedgerEra era))
+    -> Shelley.UTxO (ShelleyLedgerEra era)
 utxoFromTxOutsInRecentEra era = withConstraints era $
-    fmap (Shelley.UTxO . Map.fromList) . mapM downcast
+    Shelley.UTxO . Map.fromList . map downcast
   where
     downcast
         :: (TxIn, TxOutInRecentEra)
-        -> Either
-            ErrInvalidTxOutInEra
-            (TxIn, TxOut (ShelleyLedgerEra era))
-    downcast (i, o) = do
-        o' <- unwrapTxOutInRecentEra era o
-        pure (i, o')
+        -> (TxIn, TxOut (ShelleyLedgerEra era))
+    downcast (i, o) =
+        let
+            o' = unwrapTxOutInRecentEra era o
+        in
+            (i, o')
```

Comment:
```
    downcast
        :: (TxIn, TxOutInRecentEra)
        -> Either ErrInvalidTxOutInEra (TxIn, TxOut (ShelleyLedgerEra era))
    downcast = traverse (unwrapTxOutInRecentEra era) 
 ```

Reply:
Went with `second` in the other function instead, and the `Either` is gone now.